profiling/src/internal/observation/timestamped_observations.rs

```diff
@@ -0,0 +1,99 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present Datadog, Inc.
+
+//! Used to store timestamped observations in a compressed buffer. Assumption is that we don't need this data until
+// serialization, so it's better to pack it in while we're holding it.
+
+use super::super::LabelSetId;
+use super::super::Sample;
+use super::super::StackTraceId;
+use crate::collections::identifiable::Id;
+use crate::internal::Timestamp;
+use byteorder::{NativeEndian, ReadBytesExt};
+use lz4_flex::frame::FrameDecoder;
+use lz4_flex::frame::FrameEncoder;
+use std::io::Cursor;
+use std::io::Write;
+
+pub struct TimestampedObservations {
+    compressed_timestamped_data: FrameEncoder<Vec<u8>>,
+    sample_types_len: usize,
+}
+
+impl TimestampedObservations {
+    pub fn new(sample_types_len: usize) -> Self {
+        // Create buffer with a big capacity to avoid lots of small allocations for growing it
+        TimestampedObservations {
+            compressed_timestamped_data: FrameEncoder::new(Vec::with_capacity(1_048_576)),
+            sample_types_len: sample_types_len,
+        }
+    }
+
+    pub fn add(&mut self, sample: Sample, ts: Timestamp, values: Vec<i64>) {
+        // We explicitly turn the data into a stream of bytes, feeding it to the compressor.
+        // @ivoanjo: I played with introducing a structure to serialize it all-at-once, but it seems to be a lot of
+        // boilerplate (of which cost I'm not sure) to basically do the same as these few lines so in the end I came
+        // back to this.
+
+        let stack_trace_id = Id::into_raw_id(sample.stacktrace) as u32;
```

Comment:
Doesn't seem to work :disappointed: :

```
39 |         let stack_trace_id: u32 = Id::into_raw_id(sample.stacktrace).into();
   |                                                                      ^^^^ the trait `From<usize>` is not implemented for `u32`

```

Reply:
Ahh, there is a downcast.  We could either `.try_into().expect("error")` (since we're not returning an error yet), or a nicer solution would be to add `Into<u32>` directly onto the `StackTraceId` type.  I can help you do that if that's useful :)

Reply:
Add the following to `stack_trace.rs`

```rust
impl Into<u32> for StackTraceId  {
    fn into(self) -> u32 {
        self.0.into()
    }
}
```

Reply:
And then you can just go `sample.stacktrace.into()`

Reply:
Actually, `From` is preferred over `into`.  

```rust
impl From<StackTraceId> for u32 {
    fn from(value: StackTraceId) -> Self {
        value.0.into()
    }
}
```

Reply:
Fixed in 04b2b16eedd3f29aa68a7ac539e3fc3f9a25145a